### PR TITLE
Adds the TGS DMAPI

### PR DESCRIPTION
**Pull request by Cyberboss**
```
(u)Cyberboss:
(*)Added TGS support for hosting!
```

This just adds the basics and some event handling. It does not include the all the features available in tgs.dm such as chat messages and revision info/test merge information.

### CODE REVIEW

**Comment by ZeWaka** [comment]
lol

**Comment by Cyberboss** [comment]
Can't be cloned on TGS currently because of the submodule

In theory (if it's on github) it should work if you provide and access token for it. But you'll have to test it.

File: _stdlib/tgs.dm [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+#define TGS_INFO_LOG(message)
+
+/// Write an warning `message` to a server log.
+#define TGS_WARNING_LOG(message)
+
+/// Write an error `message` to a server log.
+#define TGS_ERROR_LOG(message)
+
+/// Get the number of connected /clients.
+#define TGS_CLIENT_COUNT
```
Comment (ZeWaka) [inline _stdlib/tgs.dm:49]: Apparently a old TGS3 thing, but unimplemented in the external config.

File: goonstation.dme [hunk 2]
```diff
@@ -52,7 +54,7 @@ var/datum/preMapLoad/preMapLoad = new
 
 //Secret Code Handling
 #ifndef FASTDMM
-#include "+secret\__secret.dme"
+// #include "+secret\__secret.dme"
```
Comment (ZeWaka) [inline goonstation.dme:57]:
```suggestion
#include "+secret\__secret.dme"
```

File: code/modules/tgs/LICENSE [hunk 3]
```diff
@@ -0,0 +1,24 @@
+The MIT License
+
+Copyright (c) 2017 Jordan Brown
```
Comment (ZeWaka) [inline code/modules/tgs/LICENSE:3]: should probably update the copyright year btw

**Pull request merged**